### ISSUE TITLE
Potential optimization that could reduce StateDB reads in Verifyfee Function

### ISSUE THREAD

**Issue by hsyodyssey** [scroll-tech/go-ethereum#588]
The [Verifyfee function](https://github.com/scroll-tech/go-ethereum/blob/441212200a3a087841eaaee01a4e9d12d31a0b56/rollup/fees/rollup_fee.go#L198-L225) is one of the most import process in the daily workflow. 

Among this function, `CalculateL1DataFee `introduces some **StateDB level reads.** According to the current logic, if the Tx fails to validate if the L2fee is insufficient, so the StateDB reads introduced by taking the lead in calculating the L1DataFee can be avoided in this scenario.

The reason for calculating `L1DataFee `first at the moment seems to be because this `CalculateL1DataFee `contains a unit that checks whether the RLP is legal or not. If this [module](https://github.com/scroll-tech/go-ethereum/blob/441212200a3a087841eaaee01a4e9d12d31a0b56/rollup/fees/rollup_fee.go#L183-L186) is can be took out, it would be possible to validate L2fee first, then L1fee, and omit the unnecessary StateDB Read mentioned above.

I can help if this is not on the working timetable.

**Comment by 0xmountaintop** [comment]
I think you are right, we can check L2fee first.

**Comment by 0xmountaintop** [comment]
so you mean, besides simply moving  `l1DataFee, err := CalculateL1DataFee(tx, state)` after l2Fee check, we can do further improvements?

**Comment by hsyodyssey** [comment]
> so you mean, besides simply moving `l1DataFee, err := CalculateL1DataFee(tx, state)` after l2Fee check, we can do further improments?

not really; switching the transaction fee verification logic should be enough so far. Just found many references on CalculateL1DataFee; refactoring it could be a little non-trival.
